src/app/systelab-translate/number-format.pipe.ts

```diff
@@ -0,0 +1,34 @@
+import { Pipe, PipeTransform } from '@angular/core';
+import { I18nService } from './i18n.service';
+import { DecimalPipe } from '@angular/common';
+
+@Pipe({
+	name: 'numberformat'
+})
+export class NumberFormatPipe implements PipeTransform {
+	constructor(protected i18nService: I18nService, protected decimalPipe: DecimalPipe) {
+	}
+
+	public transform(value: number, precision?: string, units?: string, priorSymbol?: string, defaultSymbolWhenNull?: string, ...args: string[]): string {
+
+		if (value) {
+			if (!precision) {
+				precision = '1.0-2';
+			}
+			let roundedValue = this.decimalPipe.transform(value, precision, this.i18nService.getLocale());
```

Comment:
You should protect the possible exception when precision is in a invalid format.